file_stream/file_stream.go

```diff
@@ -0,0 +1,111 @@
+package file_stream
+
+import (
+	"context"
+	"io"
+	"log"
+	"os"
+
+	"github.com/fsnotify/fsnotify"
+)
+
+type FileStream struct {
+	file_name string
+	reader    *os.File
+	watcher   *fsnotify.Watcher
+
+	// Used for get command completion notifications
+	Done chan bool
+
+	// Used to abort streaming when a client disconnects, etc
+	ctx context.Context
+}
+
+//-------------------------------------------------------------------------------------------------
+func NewFileStream(file_name string, ctx context.Context) *FileStream {
+	file, err := os.Open(file_name)
+	if err != nil {
+		log.Fatalf("Failed to open file '%s': %v", file_name, err)
+	}
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Fatalf("Failed to start a watcher: %v", err)
+	}
+
+	err = watcher.Add(file_name)
+	if err != nil {
+		log.Fatalf("Failed to add file '%s' to the watcher: %v", file_name, err)
+	}
+
+	return &FileStream{
+		file_name: file_name,
+		reader:    file,
+		watcher:   watcher,
+		ctx:       ctx,
+		Done:      make(chan bool),
+	}
+}
+
+//-------------------------------------------------------------------------------------------------
+func (s *FileStream) Close() error {
```

Comment:
What will happen if you close a stream more than once?

Reply:
It will depend on the implementation of file.Close in go. Given that they do not specify how it should behave, we'll basically inherit their behavior.

Reply:
Added some extra logic for handling double-close situations along with the logic to allow a close() call to stop any active read operations.